File: Problems/No.2269_Find_the_K_Beauty_of_a_Number/find_the_k_beauty_of_a_number.cpp
class Solution {
public:
    int divisorSubstrings(int num, int k) {
        
        // parameter
        int k_beauty = 0;
        std::string numString = std::to_string(num);

        // start computing
        for (int i=0; i<numString.length()-k+1; i++) {

            int sub_num = std::stoi(numString.substr(i, k));
            if (sub_num != 0) if (num%sub_num == 0) k_beauty++;

        }

        return k_beauty;

    }
};

File: Problems/No.2269_Find_the_K_Beauty_of_a_Number/find_the_k_beauty_of_a_number.py
class Solution(object):
    def divisorSubstrings(self, num, k):
        """
        :type num: int
        :type k: int
        :rtype: int
        """

        # parameters
        k_beauty = 0
        numString = str(num)

        # start computing
        for i in range(len(numString)-k+1):
            sub_num = int(numString[i:i+k])
            if (sub_num != 0):
                if (num%sub_num == 0):
                    k_beauty += 1

        return k_beauty
